sui/src/rpc_gateway_client.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::rest_gateway::responses::ObjectResponse;
+use crate::rpc_gateway::{
+    Base64EncodedBytes, RpcGatewayClient as RpcGateway, SignedTransaction, TransactionBytes,
+};
+use anyhow::Error;
+use async_trait::async_trait;
+use ed25519_dalek::ed25519::signature::Signature;
+use jsonrpsee::http_client::{HttpClient, HttpClientBuilder};
+use move_core_types::identifier::Identifier;
+use move_core_types::language_storage::TypeTag;
+use sui_core::gateway_state::gateway_responses::TransactionResponse;
+use sui_core::gateway_state::{GatewayAPI, GatewayTxSeqNumber};
+use sui_types::base_types::{ObjectID, ObjectRef, SuiAddress, TransactionDigest};
+use sui_types::messages::{Transaction, TransactionData};
+use sui_types::object::ObjectRead;
+use tokio::runtime::Handle;
+
+pub struct RpcGatewayClient {
+    client: HttpClient,
+}
+
+impl RpcGatewayClient {
+    pub fn new(server_url: String) -> Result<Self, anyhow::Error> {
+        let client = HttpClientBuilder::default().build(&server_url)?;
+        Ok(Self { client })
+    }
+}
+
+#[async_trait]
+impl GatewayAPI for RpcGatewayClient {
+    async fn execute_transaction(&mut self, tx: Transaction) -> Result<TransactionResponse, Error> {
+        let signature = tx.tx_signature.as_bytes();
```

Comment:
Ideally, use api from our crypto.rs to not assume static lengths etc in case we ever change signature scheme.
https://github.com/MystenLabs/sui/blob/main/sui_types/src/crypto.rs#L194

Reply:
We have sig_bytes and pub_key_bytes fn in the Signature type impl already, I will make them pub and use that instead of static lengths